### PR TITLE
Fix/overlapping tcp segments

### PR DESCRIPTION
**Pull request by urso**

### CODE REVIEW

File: packetbeat/protos/tcp/tcp.go [hunk 1]
```diff
@@ -165,9 +173,25 @@ func (tcp *Tcp) Process(id *flows.FlowID, tcphdr *layers.TCP, pkt *protos.Packet
 				}
 			}
 		}
+
+		// cut away packet overlap
+		if tcpSeqBefore(tcpStartSeq, lastSeq) {
+			delta := lastSeq - tcpStartSeq
+
+			// if 'overlap' already covered by previous packet -> return
+			if int(delta) >= len(pkt.Payload) {
```
Comment (ruflin) [inline packetbeat/protos/tcp/tcp.go]: Should we add at least a debug log message here?

**Comment by ruflin** [comment]
LGTM. Worth backporting?

File: packetbeat/protos/tcp/tcp.go (hunk 1 above)
Comment (urso) [inline packetbeat/protos/tcp/tcp.go]: debug message is ok. but this is perfectly valid behavior.

**Comment by urso** [comment]
Yeah. in presence of overlapping resends, application layer parser are likely to trip or generate panic.

File: packetbeat/protos/tcp/tcp.go (hunk 1 above)
Comment (urso) [inline packetbeat/protos/tcp/tcp.go]: rethinking this case, it's impossible to trigger this condition due to tcpSeq already compared to lastSeq and if reset, packet is dropped.

**Comment by ruflin** [comment]
I added the backport layer. There seem to be quite some packetbeat system tests that are failing ...

**Pull request merged**